src/lib/api/preview.spec.ts

```diff
@@ -2,91 +2,155 @@
  *
  * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
+ * distributed under the License is distributed on an 'AS IS' BASIS,
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
 
-import * as assert from 'assert';
-import * as sinon from 'sinon';
-import { preview } from './preview';
+import { preview, getUrl } from './preview';
 
-declare var ENV: any;
+(global as any).window = {
+  open: jest.fn(),
+};
 
-const session = ENV.session;
-const secureSession = ENV.secureSession;
-const filelink = ENV.filelink;
+const mockAppendChild = jest.fn();
+(global as any).document = {
+  createElement: jest.fn(() => {
+    return {
+      src: '',
+      width: '',
+      height: '',
+    };
+  }),
+  getElementById: jest.fn((id) => {
+    let obj;
+    if (id === 'testId') {
+      obj = {
+        appendChild: mockAppendChild,
+      };
+    }
+    return obj;
+  }),
+};
 
-describe('preview', () => {
-  let appendChildSpy: sinon.SinonSpy;
-  let createElementSpy: sinon.SinonStub;
-  let documentSpy: sinon.SinonStub;
-  let windowOpenSpy: sinon.SinonStub;
-
-  beforeEach(() => {
-    appendChildSpy = sinon.spy();
-    createElementSpy = sinon.stub(document, 'createElement').returns({});
-    documentSpy = sinon.stub(document, 'getElementById').returns({
-      appendChild: appendChildSpy,
+describe('api:upload:preview', () => {
+  describe('preview', () => {
+    it('should open a proper url with preview of a file', () => {
+      const defaultSession = {
+        'apikey': 'CmrB9kEilS1SQeHIDf3wtz',
+        'urls': {
+          'fileApiUrl': 'https://www.filestackapi.com/api/file',
+          'uploadApiUrl': 'https://upload.filestackapi.com',
+          'cloudApiUrl': 'https://cloud.filestackapi.com',
+          'cdnUrl': 'https://cdn.filestackcontent.com',
+          'pickerUrl': 'https://static.filestackapi.com/picker/1.4.4/picker.js'
+        },
+      };
+      const handle = 'testHandle';
+      preview(defaultSession, handle);
+      expect(window.open).toBeCalledTimes(1);
+      expect(window.open).toBeCalledWith('https://cdn.filestackcontent.com/preview/testHandle', 'testHandle');
     });
-    windowOpenSpy = sinon.stub(window, 'open');
-  });
-
-  afterEach(() => {
-    createElementSpy.restore();
-    windowOpenSpy.restore();
-    documentSpy.restore();
-  });
-
-  it('should throw exception when handle is not provided', () => {
-    assert.throws(() => preview(session));
-  });
-
-  it('should open window when no id specified', () => {
-    preview(session, filelink);
-    assert.ok(windowOpenSpy.calledWith(`${session.urls.cdnUrl}/preview/${filelink}`, filelink));
-  });
+    it('should open a proper url with preview of a file with src handle', () => {
+      const defaultSession = {
```

Comment:
Define default session before all tests, you dont have to repeat it in all tests

Reply:
ok